### PR TITLE
Refactor bindings

### PR DESCRIPTION
**Pull request by liamhuber**
Moving more structure methods over to sub-modules based on where our power is actually coming from.

Outstanding stuff that's making this draft:
- @zendegani: Better handling of the docstrings, i.e. with `wraps` and `wrapped_method.__doc__ = "SOME STRING\n" + wrapped_method.__doc__` 
- @sudarsan-surendralal: Moving the remaining more complex methods over to `AseFactory` where appropriate (e.g. `surface`).
- @liamhuber: making `pyiron_to_ase` (and `pyiron_to_X`) a method of `Atoms`, and similarly look into making `ase_to_pyiron` some sort of `from_ase` method.
- @liamhuber: Failing tests, or at least a first look at these failing tests.

### CODE REVIEW

**Comment by zendegani** [comment]
@liamhuber 
The errors are resolved and it is working now.
Handling/prepending of `docstring` is implemented as 
```python
wrapped_method.__doc__ = func("SOME STRING\n") + wrapped_method.__doc__
```

**Comment by jan-janssen** [comment]
```
======================================================================
1996
1997
ERROR: test_cyl_and_spherical_avg (atomistics.volumetric.test_volumetric_data.TestVolumetricData)
1998
1999
----------------------------------------------------------------------
2000
2001
2002
2003
2004
Traceback (most recent call last):
2005
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/tests/atomistics/volumetric/test_volumetric_data.py", line 68, in test_cyl_and_spherical_avg
2006
    struct = StructureFactory.ase_bulk("Al")
2007
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/generic/util.py", line 192, in decorated
2008
    return function(*args, **kwargs)
2009
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/pyiron_atomistics/atomistics/structure/factory.py", line 82, in ase_bulk
2010
    return self.ase.bulk(*args, **kwargs)
2011
AttributeError: 'str' object has no attribute 'ase'
2012

2013
----------------------------------------------------------------------
```

**Comment by liamhuber** [comment]
I'll look at the tests later tonight.

File: pyiron_atomistics/atomistics/structure/factories/ase.py [hunk 1]
```diff
@@ -36,19 +36,19 @@ def ase_wrapped_doc(f_name, chain):
...
         s.publication_add(publication_ase())
-        return ase_bulk(*args, **kwargs)
+        return ase_to_pyiron(ase_bulk(*args, **kwargs))
     bulk.__doc__ = ase_wrapped_doc('bulk','ase.build.bulk') + bulk.__doc__
 
     @wraps(ase_cut)
     def cut(self, *args, **kwargs):
         s.publication_add(publication_ase())
-        return ase_cut(*args, **kwargs)
+        return ase_to_pyiron(ase_cut(*args, **kwargs))
```
Comment (zendegani) [inline pyiron_atomistics/atomistics/structure/factories/ase.py]: I think as @sudarsan-surendralal and @niklassiemer pointed out in [#76](https://github.com/pyiron/pyiron_atomistics/pull/76#discussion_r579790057), there is no need to use `ase_to_pyiron` for `cut` and `stack`.
```suggestion
        return ase_cut(*args, **kwargs)
```

File: pyiron_atomistics/atomistics/structure/factories/ase.py [hunk 2]
```diff
@@ -36,19 +36,19 @@ def ase_wrapped_doc(f_name, chain):
...
     cut.__doc__ = ase_wrapped_doc('cut','ase.build.cut') + cut.__doc__
 
     @wraps(ase_stack)
     def stack(self, *args, **kwargs):
         s.publication_add(publication_ase())
-        return ase_stack(*args, **kwargs)
+        return ase_to_pyiron(ase_stack(*args, **kwargs))
```
Comment (zendegani) [inline pyiron_atomistics/atomistics/structure/factories/ase.py]:
```suggestion
        return ase_stack(*args, **kwargs)
```

File: pyiron_atomistics/atomistics/structure/factories/ase.py (hunk 1 above)
Comment (liamhuber) [reply pyiron_atomistics/atomistics/structure/factories/ase.py]: Yes, that was also my understanding.

**Comment by liamhuber** [comment]
> ======================================================================
> 1996
> 1997
> ERROR: test_cyl_and_spherical_avg (atomistics.volumetric.test_volumetric_data.TestVolumetricData)
> 1998
> 1999
> ----------------------------------------------------------------------
> 2000
> 2001
> 2002
> 2003
> 2004
> Traceback (most recent call last):
> 2005
>   File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/tests/atomistics/volumetric/test_volumetric_data.py", line 68, in test_cyl_and_spherical_avg
> 2006
>     struct = StructureFactory.ase_bulk("Al")
> 2007
>   File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/generic/util.py", line 192, in decorated
> 2008
>     return function(*args, **kwargs)
> 2009
>   File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/pyiron_atomistics/atomistics/structure/factory.py", line 82, in ase_bulk
> 2010
>     return self.ase.bulk(*args, **kwargs)
> 2011
> AttributeError: 'str' object has no attribute 'ase'
> 2012
> 
> 2013
> ----------------------------------------------------------------------

Right, ok I understand this now. The problem is that the test is trying to invoke `StructureFactory.ase.bulk("Al")` instead of `StructureFactory().ase.bulk("Al")`.

For now I will patch the unit test, but before we merge this into branch I'd like some feedback -- is that syntax we *want* to support? I'm inclined to say no. If you want to use a factory, instantiate one. Don't just try pulling methods off the class. In general that's a bad idea since many of our factories take instantiation arguments (like a `Project` instance). I'm open to counterpoints though.

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by liamhuber** [comment]
> Looks good to me

Cool. There's some more touch-ups I want to do. I'll try to finish them tonight and leave it all day tomorrow in a passing state for final feedback prior to merging 😃

**Pull request merged**